pyface/data_view/abstract_data_model.py

```diff
@@ -0,0 +1,284 @@
+
+        The text for column headers are returned by calling this method
+        with row as Root.
+
+        Parameters
+        ----------
+        row : sequence of int
+            The indices of the row as a sequence from root to leaf.
+        column : sequence of int
+            The indices of the column as a sequence of length 1.
+
+        Returns
+        -------
+        text : str
+            The text to display in the given row and column.
+        """
+        return str(self.get_value(row, column))
+
+    @abstractmethod
+    def set_text(self, row, column, text):
+        """ Set the text value for the row and column.
+
+        The text for column headers can be set by calling this method
+        with row as Root.
+
+        Parameters
+        ----------
+        row : sequence of int
+            The indices of the row as a sequence from root to leaf.
+        column : sequence of int
+            The indices of the column as a sequence of length 1.
+        text : str
+            The new text value for the given row and column.
+
+        Returns
+        -------
+        success : bool
+            Whether or not the value was set successfully.
+        """
+        raise NotImplementedError
+
+'''
+    @abstractmethod
+    def get_checked(self, row, column):
+        return None
+
+    @abstractmethod
+    def set_checked(self, row, column, checked):
+        return None
+
+    @abstractmethod
+    def get_color(self, row, column):
+        return None
+
+    @abstractmethod
+    def get_image(self, row, column):
+        return None
+
+    @abstractmethod
+    def get_description(self, row, column):
+        return None
+
+    # interaction methods
+
+    # XXX these should perhaps live in a separate class
+
+    def get_enabled(self, row, column):
+        """ Whether or not the given cell is enabled for user interaction
+
+        Note that if the entire control is disabled then individual cells will
+        still be disabled independent of the value returned by this method.
+
+        Parameters
+        ----------
+        row : sequence of int
+            The indices of the row as a sequence from root to leaf.
+        column : sequence of int
+            The indices of the column as a sequence of length 1.
+
+        Returns
+        -------
+        enabled : bool
+            Whether the cell allows user interaction.
+        """
+        return True
+
+    def get_editable(self, row, column):
+        return False
+
+    @abstractmethod
+    def get_column_editor(self, column):
+        """ Return editor information for a column
+        """
+        # XXX needs to be a richer object
+        # eg should have bounds for spinbox, choices for Combo etc.
+        return "text"
+
+    @abstractmethod
+    def get_cell_editor(self, row, column):
+        """ Return editor information for the row and column.
+        """
+        return self.get_column_type(column)
```

Comment:
> Do you already have an idea for how the combobox / spinbox / anything else going to fit in this new table and tree widget?

These two look like the relevant methods. But that's about as much as this goes. We probably need QAbstractItemDelegate for these, right?